From dfeng@redhat.com  Mon Jul 27 09:39:27 2009
From: Xiaotian Feng <dfeng@redhat.com>
Date: Fri, 24 Jul 2009 17:31:41 +0800
Subject: driver core: sysdev: do not send KOBJ_ADD uevent if kobject_init_and_add fails
To: gregkh@suse.de
Cc: linux-kernel@vger.kernel.org, Xiaotian Feng <dfeng@redhat.com>
Message-ID: <1248427901-30992-1-git-send-email-dfeng@redhat.com>


If kobject_init_and_add fails, sysdev_register should not send KOBJ_ADD
uevent to userspace.

Signed-off-by: Xiaotian Feng <dfeng@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/sys.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/base/sys.c
+++ b/drivers/base/sys.c
@@ -275,9 +275,9 @@ int sysdev_register(struct sys_device *s
 				drv->add(sysdev);
 		}
 		mutex_unlock(&sysdev_drivers_lock);
+		kobject_uevent(&sysdev->kobj, KOBJ_ADD);
 	}
 
-	kobject_uevent(&sysdev->kobj, KOBJ_ADD);
 	return error;
 }
 
